homeassistant/components/life360/device_tracker.py

```diff
@@ -0,0 +1,624 @@
+    vol.Optional(CONF_ADD_ZONES): vol.Any(
+        AZ_ONCE, vol.All(cv.time_period, vol.Range(min=MIN_ZONE_INTERVAL))),
+    vol.Optional(CONF_CIRCLES): _EXCL_INCL_LOWER_STRING_LIST,
+    vol.Optional(CONF_DRIVING_SPEED): vol.Coerce(float),
+    vol.Optional(CONF_ERROR_THRESHOLD): _THRESHOLD,
+    vol.Optional(CONF_FILENAME, default=DEFAULT_FILENAME): cv.string,
+    vol.Optional(CONF_HOME_PLACE, default=DEFAULT_HOME_PLACE): cv.string,
+    vol.Optional(CONF_MAX_GPS_ACCURACY): vol.Coerce(float),
+    vol.Optional(CONF_MAX_UPDATE_WAIT): vol.All(
+        cv.time_period, cv.positive_timedelta),
+    vol.Optional(CONF_MEMBERS): _EXCL_INCL_SLUG_LIST,
+    vol.Optional(CONF_PLACES): _EXCL_INCL_LOWER_STRING_LIST,
+    vol.Optional(CONF_PREFIX, default=DEFAULT_PREFIX):
+        vol.All(vol.Any(None, cv.string), _prefix),
+    vol.Optional(CONF_SHOW_AS_STATE, default=[]): vol.All(
+        cv.ensure_list, [vol.In(SHOW_AS_STATE_OPTS)]),
+    vol.Optional(CONF_WARNING_THRESHOLD): _THRESHOLD,
+})
+
+PLATFORM_SCHEMA = vol.All(LIFE360_SCHEMA, _thresholds)
+
+
+def _include_name(filter_dict, name):
+    if not name:
+        return False
+    if not filter_dict:
+        return True
+    name = name.lower()
+    if filter_dict['include']:
+        return name in filter_dict['list']
+    return name not in filter_dict['list']
+
+
+def _exc_msg(exc):
+    return '{}: {}'.format(exc.__class__.__name__, str(exc))
+
+
+def _setup_zone_updating(hass, config, api):
+    from life360 import Life360Error
+
+    add_zones = config.get(CONF_ADD_ZONES)
+    if not add_zones:
+        return
+
+    Place = namedtuple(
+        'Place', [ATTR_NAME, ATTR_LATITUDE, ATTR_LONGITUDE, ATTR_RADIUS])
+
+    def get_places(api, circles_filter, places_filter):
+        errs = 0
+        while True:
+            places = set()
+            try:
+                for circle in api.get_circles():
+                    circle_id = circle['id']
+                    if not (circle_id
+                            and _include_name(circles_filter, circle['name'])):
+                        continue
+                    for place in api.get_circle_places(circle_id):
+                        place_name = place['name']
+                        if not _include_name(places_filter, place_name):
+                            continue
+                        places.add(Place(place_name,
+                                         float(place['latitude']),
+                                         float(place['longitude']),
+                                         float(place['radius'])))
+            except (Life360Error, KeyError, TypeError, ValueError) as exc:
+                errs += 1
+                if errs >= 3:
+                    _LOGGER.error('get_places: %s', _exc_msg(exc))
+                    return None
+            else:
+                return places
+
+    def zone_from_place(place, entity_id=None):
+        zone = Zone(hass, *place, None, DEFAULT_PASSIVE)
```

Comment:
I'm not sure if this is allowed, creating an entity of a different domain. Is there some precedence?

Reply:
I don't know about a precedence, I only know about the need.

Life360 has the concept of Places, which is basically the same idea as zones in HA. The first issue arose due to the fact that each have different definitions of "Home." Users discovered that HA and Life360 did not always agree when a device was "home." So I added a feature to the custom component to update HA's home zone to be the same location & radius as Life360's "Home Place." Then others discovered it was extra, error prone work to try and keep HA zones to be defined the same as Life360's Places (like work, school, etc.) So the feature was extended to create HA zones automatically from Life360 Places. That way a user doesn't have to try to manually keep the two in sync. And there's flexibility to still have HA unique zones and Life360 unique Places, too.

I'll grant that maybe the implementation wasn't what the core team might expect or want, and I'm willing to change it, of course. But the bottom line is this feature was requested by the users of my custom component, and many find it very useful indeed.

Maybe the zone implementation needs to be extended to include a service, or maybe a dispatcher event, to update/create zones on the fly. (Or does this capability already exist and I missed it?) The system as it stands seems to handle zones changing dynamically just fine.

Reply:
To answer your question about a precedence, yes, owntracks is creating zones as well:

https://github.com/home-assistant/home-assistant/blob/f386088defcdbae4eb5cfbda741c606e34bb45c2/homeassistant/components/owntracks/device_tracker.py#L259-L277

Reply:
Good! Let's hear what core thinks in light of this.